statsdbinterface/rankings.py

```diff
@@ -114,50 +114,40 @@ def players_by_games(days):
 
 @cached(60)
 def modes_by_games(days):
-    re = redeclipse.versions.default
     first_game = first_game_in_days(days)
-    modes = extmodels.Mode.mode_list()
+    modes = models.Mode.query \
+        .filter(models.Mode.name.notin_(['edit', 'demo'])) \
+        .all()
     ret = []
     for mode in modes:
-        m = {
-            'name': mode,
-            'longname': re.modestr[re.modes[mode]],
-            'games': 0,
-            }
-        m['games'] = models.Game.query.filter(
-            db.func.re_mode(models.Game.id, mode),
-            models.Game.id >= first_game
-            ).count()
+        m = {'name': mode.name,
+             'longname': mode.longname,
+             'games': mode.games.filter(models.Game.id >= first_game)
+                                .count()}
         ret.append(m)
     return sorted(ret, key=lambda m: m['games'], reverse=True)
```

Comment:
@zaquest (after edit) you're right, according to the [docs](https://docs.python.org/3/library/functions.html#filter) it'll always return a list. I probably thought of Python's `filter()` (and some other functions) that return an iterator now.

Actually, it's pretty obvious why an iterator wouldn't be more efficient here. Sorting algorithms don't work item by item, Timsort (the Python sorting algorithm) works in-place anyway.

What *could* further be improved here though is to get rid of `sorted()`. Sorted creates a copy of the list you pass to it, and returns that. Here, we could just sort in-place though, as the list may be modified before returning it.

So, I'd use `ret.sort(key=lambda m: m["games"])` and then `return ret`.